uber/models.py

```diff
@@ -704,13 +704,14 @@ def get_next_badge_num(self, badge_type):
             badge_type = get_real_badge_type(badge_type)
 
             new_badge_num = self.auto_badge_num(badge_type)
+            lower_bound = c.BADGE_RANGES[badge_type][0]
+            upper_bound = c.BADGE_RANGES[badge_type][1]
             # Adjusts the badge number based on badges in the session
             for attendee in [m for m in chain(self.new, self.dirty) if isinstance(m, Attendee)]:
-                if attendee.badge_num is not None \
-                        and c.BADGE_RANGES[badge_type][0] <= attendee.badge_num <= c.BADGE_RANGES[badge_type][1]:
-                    new_badge_num = max(self.auto_badge_num(badge_type), 1 + attendee.badge_num, new_badge_num)
```

Comment:
Huh, I probably thought this was necessary to take care of all the dirty/new badges in a session... but if the unit tests pass, it should be OK. Neat!